libs/application/templates/public-debt-payment-plan/src/fields/PaymentPlan/PaymentPlan.tsx

```diff
@@ -25,58 +27,104 @@ import { PlanSlider } from '../components/PlanSlider/PlanSlider'
   const answers = application.answers as PublicDebtPaymentPlan
   const index = field.defaultValue as number
   // Assign a payment to this screen by using the index of the step
-  const payment = externalData.paymentPlanList?.data[index]
+  const payment = externalData.paymentPlanPrerequisites?.data?.debts[index]
+  // Geta min/max month and min/max payment data
+  const initialMinMaxData = externalData.paymentPlanPrerequisites?.data?.allInitialSchedules.find(
+    (x) => x.scheduleType === payment?.type,
+  )
   // Locate the entry of the payment plan in answers.
   const entryKey = getPaymentPlanKeyById(
     answers.paymentPlans,
-    payment?.id || '',
+    payment?.type || '',
   )
   // If no entry is found, find an empty entry to assign to this payment
   const answerKey = (entryKey ||
     getEmptyPaymentPlanEntryKey(
       answers.paymentPlans,
     )) as keyof typeof answers.paymentPlans
 
-  if (!answerKey) {
-    // There is no entry available for this plan
-    // The user can not continue the application
-    // TODO: Better UX and error logging for this
-    return <div>No more available entries in schema</div>
-  }
   const entry = `paymentPlans.${answerKey}`
-  const currentAnswers = answers.paymentPlans[answerKey]
+  const currentAnswers = answers.paymentPlans
+    ? answers.paymentPlans[answerKey]
+    : undefined
+
+  const [paymentMode, setPaymentMode] = useState<PaymentModeState | undefined>(
+    currentAnswers?.paymentMode,
+  )
+
+  const getDistributionCallback = useCallback(
+    async ({
+      monthAmount = null,
+      monthCount = null,
+      totalAmount,
+      scheduleType,
+    }: GetScheduleDistributionInput) => {
+      const { data } = await getDistribution({
+        input: {
+          monthAmount,
+          monthCount,
+          totalAmount,
+          scheduleType,
+        },
+      })
+
+      return data
+    },
+    [getDistribution],
+  )
 
   const {
     debouncedAmount,
     debouncedMonths,
     setAmount,
     setMonths,
-    // eslint-disable-next-line react-hooks/rules-of-hooks
   } = useDebouncedSliderValues(currentAnswers)
-  // eslint-disable-next-line react-hooks/rules-of-hooks
-  const { isLoading, data: paymentPlanResults } = useMockPaymentPlan(
-    userInfo?.profile.nationalId,
-    payment?.type,
+
+  useEffect(() => {
+    if (payment) {
+      setIsLoading(true)
+      getDistributionCallback({
+        monthAmount: paymentMode === AMOUNT ? debouncedAmount : null,
+        monthCount: paymentMode === MONTHS ? debouncedMonths : null,
+        totalAmount: payment.totalAmount,
+        scheduleType: payment.type,
+      })
+        .then((response) => {
+          setDistributionData(response?.paymentScheduleDistribution || null)
+          setIsLoading(false)
+        })
+        .catch((error) => {
+          // Do something?
+          console.log('An error occured fetching payment distribution: ', error)
```

Comment:
Should we log this to sentry or datadog?

Reply:
Change to console.error for sentry to catch it 👍